flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/event/JdbcAutoScalerEventHandler.java

```diff
@@ -104,6 +134,43 @@ public void handleScalingEvent(
         }
     }
 
+    @Override
+    public void close() {
+        if (Objects.nonNull(scheduledEventHandlerCleaner)
+                && !scheduledEventHandlerCleaner.isShutdown()) {
+            scheduledEventHandlerCleaner.shutdownNow();
+        }
+    }
+
+    @VisibleForTesting
+    void cleanExpiredEvents() {
+        try {
+            JdbcEventInteractor.ExpiredEventsResult expiredResult =
+                    jdbcEventInteractor.queryExpiredEventsAndMaxId(eventHandlerTtl);
+            if (Objects.isNull(expiredResult)) {
+                log.warn("No expired event handlers queried at {}", new Date());
+                return;
+            }
+            var batch = 4098;
+            var startId = expiredResult.minId;
+            var endId = Math.min(startId + batch, expiredResult.maxId);
+            while (startId <= expiredResult.maxId) {
+                jdbcEventInteractor.deleteExpiredEventsByIdRange(startId, endId);
+                startId += batch;
+                endId = Math.min(endId + batch, expiredResult.maxId);
+                log.debug("Deleted expired event handler records by batch successfully.");
+                Thread.sleep(10L);
+            }
```

Comment:
It seems the loop is pretty easier than while loop, and we don't need to introduce a series of variables.

```suggestion
            var batch = 4098;
            for (var startId = expiredResult.minId; startId <= expiredResult.maxId; startId += batch) {
                jdbcEventInteractor.deleteExpiredEventsByIdRange(startId, Math.min(start + batch, expiredResult.maxId));
                log.debug("Deleted expired event handler records by batch successfully.");
                Thread.sleep(10L);
            }
```